src/cc2olx/qti.py

```diff
@@ -0,0 +1,320 @@
+import urllib.parse
+import xml.dom.minidom
+
+from collections import OrderedDict
+
+from lxml import html, etree
+from html import unescape
+
+from cc2olx import filesystem
+
+# problem types
+MULTIPLE_CHOICE = "cc.multiple_choice.v0p1"
+MULTIPLE_RESPONSE = "cc.multiple_response.v0p1"
+FILL_IN_THE_BLANK = "cc.fib.v0p1"
+ESSAY = "cc.essay.v0p1"
+BOOLEAN = "cc.true_false.v0p1"
+PATTERN_MATCH = "cc.pattern_match.v0p1"
+
+
+class QtiError(Exception):
+    """
+    Exception type for Qti parsing/conversion errors.
+    """
+
+
+class QtiExport:
+    """
+    Contains methods for processing and conversion
+    IMS Question & Test Interoperability (QTI) <= v1.2 into OLX markup
+    """
+
+    WEB_RESOURCES_DIR_ALIAS = '$IMS-CC-FILEBASE$'
+    WIKI_CONTENT_DIR_ALIAS = '$WIKI_REFERENCE$'
+
+    def __init__(self, root_xml_doc):
+        self.doc = root_xml_doc
+
+    def create_qti_node(self, details):
+        """
+        Creates OLX xml node, that represents content of unit with problems.
+
+        Args:
+            details: list of dictionaries, where each contains data to
+                render problem.
+        """
+
+        problems = []
+
+        for problem_data in details:
+            cc_profile = problem_data['cc_profile']
+            create_problem = self._problem_creators_map.get(cc_profile)
+
+            if create_problem is None:
+                raise QtiError("Unknown cc_profile: \"{}\"".format(problem_data['cc_profile']))
+
+            problem = create_problem(problem_data)
```

Comment:
Should we catch/log exceptions from this call, so we can continue processing if errors occur?

Reply:
I think we should let it fail, @azarembok because problems in the same assessment can refer one to another, and description of one problem can depend on description of other, like:

Problem 1: simplify equation X
Problem 2: given equation from previous problem, do Y

So catching an exception can in theory lead to creation of assessment, that lacks some necessary data.

Reply:
makes sense, thanks!